docs/concepts.md

```diff
@@ -0,0 +1,12 @@
+### Federation V2 Concepts
+
+| Concept | Description |
+|-----------|-------------|
+| Federate | Federating a set of Kubernetes clusters means, effectively creating a common interface to the pool of these federated clusters which can be used to deploy kubernetes applications across the same. |
+| Federation | Kubernetes Cluster Federation enables users to federate multiple Kubernetes clusters for resources distribution, service discovery, high availability etc across multiple clusters. |
+| Host Cluster | A cluster which is used to expose the federation API and run the federation control plane. |
+| Member Cluster | A cluster which is registered with the federation API and that federation controllers have authentication credentials for. The Host Cluster can also be a Member Cluster. |
```

Comment:
You may want to explain registration since you use it in the description. Maybe create a new row that explains the term registration.

Reply:
Done